vcpkg-configuration.json

```diff
@@ -0,0 +1,10 @@
+{
+  "overlay-ports": [
+    "overlay/ports",
+    "overlay/osx",
+    "overlay/windows"
+  ],
```

Comment:
I don't think this will be always the case.
Let's imagine we need to stick with an older version of a library for some reasons for one target only.

With this current approach, all targets will pick this up.

I can imagine to fix this in the vcpkg.json of each package, but this fiddling is a maintenance burden IMHO

If we decide to continue the proposed route, I would then merge everything to the ports folder to not surprise other contributors.

Reply:
I'd say moving everything to `overlay/ports` would make sense and be in line with what upstream does too (i.e. rely on vcpkg's platform mechanism rather than scoping ports on a file-level)

Reply:
Using different versions of a library for different platforms is IMO not a good idea anyway and would increase the risk of bugs being reproducible on certain platforms only. But since we're already in patch-level territory for some of the dependencies, maybe we're already past that.

Reply:
OK we may try it. Than please move everything to "overlay/ports"